framework/console/widgets/Table.php

```diff
@@ -0,0 +1,323 @@
+            $this->_chars['left'],
+            $this->_chars['middle'],
+            $this->_chars['right']
+        );
+
+        foreach ($this->_rows as $row) {
+            $buffer .= $this->renderSeparator(
+                $this->_chars['left-mid'],
+                $this->_chars['mid-mid'],
+                $this->_chars['mid'],
+                $this->_chars['right-mid']
+            );
+            $buffer .= $this->renderRows($row,
+                $this->_chars['left'],
+                $this->_chars['middle'],
+                $this->_chars['right']);
+        }
+
+        $buffer .= $this->renderSeparator(
+            $this->_chars['bottom-left'],
+            $this->_chars['bottom-mid'],
+            $this->_chars['bottom'],
+            $this->_chars['bottom-right']
+        );
+
+        return $buffer;
+    }
+
+    /**
+     * @param array $row
+     * @param string $spanLeft
+     * @param string $spanMiddle
+     * @param string $spanRight
+     * @return string the generated row
+     * @see \yii\console\widgets\Table::render()
+     */
+    protected function renderRows(array $row, $spanLeft, $spanMiddle, $spanRight)
+    {
+        $size = $this->_columnWidths;
+        $rowsPerCell = array_map(function ($size, $columnWidth) {
+                if (is_array($columnWidth)) {
+                    $rows = 0;
+                    foreach ($columnWidth as $width) {
+                        $rows += ceil($width / ($size - 2));
+                    }
+                    return $rows;
+                }
+                return ceil($columnWidth / ($size - 2));
+            },
+            $size, array_map(function($val) {
+                    if (is_array($val)) {
+                        $encodings = array_fill(0, count($val), Yii::$app->charset);
+                        return array_map('mb_strwidth', $val, $encodings);
+                    }
+                    return mb_strwidth($val, Yii::$app->charset);
+                },
+                $row
+            )
+        );
+        $buffer = '';
+        for ($i = 0, $max = max($rowsPerCell); $i < $max; $i++) {
+            $buffer .= $spanLeft . ' ';
+            foreach ($row as $index => $cell) {
+                $prefix = '';
+                if ($index !== 0) {
+                    $buffer .= $spanMiddle . ' ';
+                }
+                if (is_array($cell)) {
+                    if (empty($finalChunk[$index])) {
+                        $finalChunk[$index] = '';
+                        $start = 0;
+                        $prefix = $this->_listPrefix;
+                        if (!isset($arrayPointer[$index])) {
+                            $arrayPointer[$index] = 0;
+                        }
+                    } else {
+                        $start = mb_strwidth($finalChunk[$index], Yii::$app->charset);
+                    }
+                    $chunk = mb_substr($cell[$arrayPointer[$index]], $start, $size[$index] - 4 , Yii::$app->charset);
```

Comment:
`$arrayPointer` is not properly initialized before using it. I can't say it will cause any issues but it may to.